cloudinit/util.py

```diff
@@ -68,17 +68,31 @@
                    ['running-in-container'],
                    ['lxc-is-container'])
 
-PROC_CMDLINE = None
+_CACHED_RESPONSES = {}
 
-_LSB_RELEASE = {}
 
+def memoize(f):
```

Comment:
Instead of implementing this ourselves, should we be using [`functools.lru_cache`](https://docs.python.org/3/library/functools.html#functools.lru_cache)?

We could shim in a no-op decorator for Python 2:

```py
try:
    from functools import lru_cache
except ImportError:
    def lru_cache():
        def wrapper(f):
            return f
        return wrapper
```

(Note that I haven't tested this code, it's just to illustrate my suggestion.)

Reply:
how much harder will `lru_cache()` be to mock than what we have rn?

Reply:
by calling `wrapped_function.cache_info()`

Reply:
Mocking replaces the symbol in the target module, so this shouldn't affect mocking at all. I feel like I might not be fully understanding your question, though, so if that answer doesn't make sense could you expand on it, please?